### ISSUE TITLE
pkg: deploy model or dataset

### ISSUE THREAD

**Issue by dmpetrov** [iterative/dvc#1768]
Currently `dvc pkg install` instantiates data artifacts in a current repository and cannot work outside of a repo.

In some cases, like production model deployment scenario, a repository is not needed. The only important thing is a data file content without any meta files (aka dvc-files).

We need to implement an option (like `--flatten` or `--deploy`) for this out-of-repository scenario.

Model deployment example:
```
$ mkdir /mnt/deploy
$ ls -a /mnt/deploy
.   ..
$ dvc pkg install --flatten -s model.pkl http://github.com/dmpetrov/new_tag_classifier /mnt/deploy/
$ ls -a /mnt/deploy
.   ..    model.pkl 
```

Download a dataset example:
```
$ dvc pkg install --flatten http://github.com/dmpetrov/imagenet .
$ ls -a
.   ..    imagenet/ 
$ du -sh imagenet
118G
```

Possible option names:
1. `--flatten`, `-f` - it might conflict with the current `-f FILE`, `--file FILE`.
2. `--deploy`, `-d`

**Comment by snagacarl** [comment]
For your command `dvc pkg install --flatten -s model.pkl http://github.com/dmpetrov/new_tag_classifier /mnt/deploy/` how would the command specify the specific version of the `model.pkl`?  Should it be included in the Git URL (`http://github.com/dmpetrov/new_tag_classifier#v1.0`) or maybe via arguments (`--tag v1.0` or `--branch prod`)?

**Comment by efiop** [comment]
Fixed by #2012 . Example command (can be ran outside of dvc repo):
```
dvc pkg get https://github.com/iterative/dvc scripts/innosetup/dvc.ico
```
